node.go

```diff
@@ -0,0 +1,17 @@
+package iface
+
+import (
+	"github.com/ipfs/go-dagwriter"
+	"github.com/ipfs/go-fetcher"
+)
+
+// NodeAPI provides an interface for working directly with IPLD nodes
+// through go-ipld-prime. It is a combination of DagWritingService (for writes)
+// and Fetcher (for reads)
+type NodeAPI interface {
+	// fetcher.Factory provides the interface to get new dag fetchers
+	fetcher.Factory
+
+	// DagWritingService implements methods to write dags
+	dagwriter.DagWritingService
+}
```

Comment:
@warpfork do you mind taking a look on mostly if you're happy with the interfaces here being the things we expose to people who would otherwise be using the old DAG interfaces?

Reply:
Per [other comment thread](https://github.com/ipfs/interface-go-ipfs-core/pull/67/files#r680273675), are we just going to walk this back for now, or push it to future work?

I think I agree with the discussion in that thread that we just haven't put enough design cycles behind this to really commit to it being a public surface area that we'll promise to maintain, just yet.

Reply:
Yep, we're going to walk it back for now.